src/server/tiered_storage.cc

```diff
@@ -271,14 +271,15 @@ template util::fb2::Future<size_t> TieredStorage::Modify(DbIndex dbid, std::stri
                                                          const PrimeValue& value,
                                                          std::function<size_t(std::string*)> modf);
 
-void TieredStorage::Stash(DbIndex dbid, string_view key, PrimeValue* value) {
-  CHECK(!value->IsExternal() && !value->HasIoPending());
+bool TieredStorage::TryStash(DbIndex dbid, string_view key, PrimeValue* value) {
+  if (!ShouldStash(*value))
+    return false;
```

Comment:
please copy the `CHECK`

Reply:
ShouldStash checks all conditions of the CHECK